package.json

```diff
@@ -13,5 +13,8 @@
     "tests": "./scripts/start-containers-for-tests.sh",
     "analytics": "docker compose -f ./docker-compose.yml -f ./docker-compose.local.yml --profile mock-services --profile analytics  up -d --quiet-pull",
     "logs": "docker compose logs --tail 30 -f"
+  },
+  "devDependencies": {
+    "typescript": "5.4.5"
```

Comment:
My VSCode version of typescript was causing some issues with VSCode refactoring tools, so I added this at the root level to allow me to switch to a lower version of TS. 

Sources: https://github.com/microsoft/TypeScript/issues/58897#issuecomment-2220808697 and https://github.com/yarnpkg/berry/issues/4788#issuecomment-1767059657

If anyone knows of a better way to get VSCode to pick up a new Typescript version, let me know! I tried `pnpm install -g`...

Reply:
Strange one, haven't hit this before but happy to think on it a bit (shouldn't block merging this!). 

One thing I don't quite understand here though - why set version `5.4.5` here? I think I'd expect this to be `5.4.3` which is our lowest version of TS specified throughout this repo? (eg editor, scripts, e2e/tests/api-driven all on `5.4.3`; api `5.5.2`; and e2e root `5.5.3`?)

Reply:
Good catch, I blithely followed the issue comments without thinking about what version would actually be appropriate! 😶‍🌫️ Switched to the lowest version now.